prime-dotnet-webapi/Controllers/SitesController.cs

```diff
@@ -831,35 +828,56 @@ public async Task<ActionResult> SendSiteReviewedNotificationEmail(int siteId, Fr
         [ProducesResponseType(StatusCodes.Status401Unauthorized)]
         [ProducesResponseType(StatusCodes.Status403Forbidden)]
         [ProducesResponseType(typeof(ApiMessageResponse), StatusCodes.Status404NotFound)]
-        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status200OK)]
         public async Task<ActionResult> ApproveSite(int siteId)
         {
-            var site = await _communitySiteService.GetSiteAsync(siteId);
-            if (site == null)
+            if (!await _siteService.SiteExistsAsync(siteId))
             {
                 return NotFound($"Site not found with id {siteId}");
             }
-            if (!SiteStatusStateEngine.AllowableStatusChange(SiteRegistrationAction.Approve, site.Status))
+
+            var status = await _siteService.GetSiteStatusAsync(siteId);
+            if (!SiteStatusStateEngine.AllowableStatusChange(SiteRegistrationAction.Approve, status))
             {
                 return BadRequest("Action could not be performed.");
             }
 
-            var updatedSite = await _siteService.ApproveSite(siteId);
-            if (site.ActiveBeforeRegistration)
+            // TODO: This is the only difference in path between Community Site and Health Authority Site
+            // As well maybe we should try/catch email errors so failure on sending an email doesn't fail
+            // the call
+            if (await _communitySiteService.SiteExistsAsync(siteId))
             {
-                await _emailService.SendSiteActiveBeforeRegistrationAsync(site.Id, site.Organization.SigningAuthority.Email);
+                var pec = await _siteService.GetSitePecAsync(siteId);
+                if (pec == null)
+                {
+                    return BadRequest("Site approval requires a site ID/PEC code.");
+                }
+
+                await _siteService.ApproveSite(siteId);
+
+                var communitySite = await _communitySiteService.GetSiteAsync(siteId);
+
+                if (communitySite.ActiveBeforeRegistration)
+                {
+                    await _emailService.SendSiteActiveBeforeRegistrationAsync(siteId);
```

Comment:
This method was updated to handle `CommunitySite` vs `HealthAuthoritySite` but here it is only being called for `CommunnitySite`s

Reply:
I can revert the work to this call so it's just still a community site.